specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+        }
+      }
+    },
+    "CloudHsmCluster": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CloudHsmClusterProperties",
+          "description": "Properties of the Cloud HSM Cluster"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/CloudHsmClusterResource"
+        }
+      ],
+      "description": "Resource information with extended details.",
+      "type": "object"
+    },
+    "CloudHsmClusterResource": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Cloud HSM Cluster Resource",
+      "type": "object",
+      "x-ms-azure-resource": true
+    },
+    "CloudHsmClusterProperties": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's provisioningState",
+          "enum": [
+            "Provisioning",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's auto-generated Domain Name Label Scope"
+        },
+        "securityDomain": {
+          "$ref": "#/definitions/CloudHsmClusterSecurityDomainProperties"
+        },
+        "hsms": {
+          "type": "array",
+          "description": "An array of Cloud HSM Cluster's HSMs",
+          "x-ms-identifiers": [
+            "fqdn"
+          ],
+          "items": {
+            "$ref": "#/definitions/CloudHsmProperties"
+          }
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster public network access"
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "description": "List of private endpoint connection resources",
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Cloud HSM Cluster status message",
+          "readOnly": true
+        },
+        "cloudHsmClusterRestoreProperties": {
+          "$ref": "#/definitions/CloudHsmClusterRestoreProperties"
+        }
```

Comment:
We should also have cloudHsmClusterBackupProperties after this

Reply:
updated